pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -741,31 +730,46 @@ func (r *ReconcileClusterPool) createRandomNamespace(clp *hivev1.ClusterPool) (*
 	return ns, err
 }
 
+func getClustersToDelete(cds *cdCollection, deletionsNeeded int, logger log.FieldLogger) []*hivev1.ClusterDeployment {
```

Comment:
A comment about the logic would be helpful here

Reply:
Totally. Will do.